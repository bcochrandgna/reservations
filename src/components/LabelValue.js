// @flow
import React from 'react';
import { View, Text } from 'react-native';
import styles from '../styles/components/LabelValue';

type Props = {
  label: string,
  value: string,
};

const LabelValue = (props: Props) => {
  const { label, value } = props;

  return (
    <View style={styles.container}>
      <Text style={styles.labelText}>{`${label}:`}</Text>
      <Text style={styles.valueText}>{value}</Text>
    </View>
  );
};

export default LabelValue;
