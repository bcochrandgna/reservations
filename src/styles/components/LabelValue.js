import { StyleSheet } from 'react-native';
import { standardMargin, standardBold } from '..';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  labelText: {
    fontWeight: standardBold,
  },
  valueText: {
    marginLeft: standardMargin,
  },
});
