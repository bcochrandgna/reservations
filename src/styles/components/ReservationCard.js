import { StyleSheet } from 'react-native';
import {
  borderColor,
  normalTextColor,
  mediumTextSize,
  titleTextColor,
  cardBackgroundColor,
  standardMargin,
  standardPadding,
  cardShadedColor,
} from '..';

export default StyleSheet.create({
  container: {
    borderWidth: 1,
    borderColor,
    marginHorizontal: standardMargin,
    marginBottom: standardMargin,
    backgroundColor: cardBackgroundColor,
  },
  dateText: {
    color: normalTextColor,
    marginLeft: standardMargin,
  },
  titleText: {
    fontSize: mediumTextSize,
    color: titleTextColor,
  },
  titleContainer: {
    padding: standardPadding,
  },
  bottomContainer: {
    marginTop: standardMargin,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: cardShadedColor,
    padding: standardPadding,
  },
});
