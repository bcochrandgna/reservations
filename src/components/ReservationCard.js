// @flow
import React from 'react';
import { View, Text } from 'react-native';
import moment from 'moment';
import styles from '../styles/components/ReservationCard';
import i18n from '../helpers/i18n';
import LabelValue from './LabelValue';
import { Reservation } from '../types/Reservation';

type Props = {
  reservation: Reservation,
};

const ReservationCard = (props: Props) => {
  const { reservation } = props;

  if (!reservation || !reservation.name) return null;

  const formattedArrivalDate = reservation.arrivalDate
    && moment(new Date(reservation.arrivalDate)).format('dddd, MMMM Do YYYY');
  const formattedDepartureDate = reservation.departureDate
    && moment(new Date(reservation.departureDate)).format('dddd, MMMM Do YYYY');

  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.titleText}>{reservation.name}</Text>
        <Text>{reservation.hotelName}</Text>
      </View>
      <View style={styles.bottomContainer}>
        <View>
          <LabelValue label={i18n.t('arrival_date')} value={formattedArrivalDate} />
          <LabelValue label={i18n.t('departure_date')} value={formattedDepartureDate} />
        </View>
        <View />
      </View>
    </View>
  );
};

export default ReservationCard;
