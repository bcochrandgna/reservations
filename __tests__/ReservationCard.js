import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import ReservationCard from '../src/components/ReservationCard';
import LabelValue from '../src/components/LabelValue';

describe('ReservationCard', () => {
  it('renders correctly with null props', () => {
    const wrapper = shallow(<ReservationCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with reservation', () => {
    const reservation = {
      name: 'Test Name',
      hotelName: 'Test Hotel Name',
      arrivalDate: '03/01/2019',
      departureDate: '03/02/2019',
    };
    const wrapper = shallow(<ReservationCard reservation={reservation} />);
    const TextList = wrapper.find(Text);
    const LabelValueList = wrapper.find(LabelValue);

    expect(TextList).toHaveLength(2);
    expect(LabelValueList).toHaveLength(2);

    expect(
      TextList.at(0)
        .children()
        .text(),
    ).toEqual('Test Name');

    expect(
      TextList.at(1)
        .children()
        .text(),
    ).toEqual('Test Hotel Name');

    expect(LabelValueList.at(0).prop('label')).toEqual('Arrival');
    expect(LabelValueList.at(1).prop('label')).toEqual('Departure');
    expect(LabelValueList.at(0).prop('value')).toEqual('Friday, March 1st 2019');
    expect(LabelValueList.at(1).prop('value')).toEqual('Saturday, March 2nd 2019');

    expect(wrapper).toMatchSnapshot();
  });
});
