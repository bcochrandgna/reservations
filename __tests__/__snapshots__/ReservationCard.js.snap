// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationCard renders correctly with null props 1`] = `""`;

exports[`ReservationCard renders correctly with reservation 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#dddddd",
      "borderWidth": 1,
      "marginBottom": 5,
      "marginHorizontal": 5,
    }
  }
>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 16,
        }
      }
    >
      Test Name
    </Text>
    <Text>
      Test Hotel Name
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#f0f0f0",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 5,
        "padding": 5,
      }
    }
  >
    <View>
      <LabelValue
        label="Arrival"
        value="Friday, March 1st 2019"
      />
      <LabelValue
        label="Departure"
        value="Saturday, March 2nd 2019"
      />
    </View>
    <View />
  </View>
</View>
`;
